api/api.go

```diff
@@ -125,9 +125,10 @@ func InitServer(ctx context.Context, params APIParams) (Server, error) {
 	if err != nil {
 		return Server{}, errors.Wrap(err, "failed to init lassie")
 	}
-	endpointFinder, err := endpointfinder.NewEndpointFinder(replication.MinerInfoFetcher{
+	finderCfg := endpointfinder.EndpointFinderConfig{LruSize: 128, ErrorLruSize: 128, ErrorLruTimeout: time.Minute * 5}
```

Comment:
Consider using option parameters, or having a function that returns a config that is populated with default values.